src/commonMain/kotlin/com/charleskorn/kaml/YamlObjectInput.kt

```diff
@@ -36,11 +36,27 @@ internal class YamlObjectInput(map: YamlMap, yaml: Yaml, context: SerializersMod
 
             val currentEntry = entriesList[nextIndex]
             currentKey = currentEntry.key
-            val fieldDescriptorIndex = descriptor.getElementIndex(propertyName)
+
+            val pairedPropertyNames = (0 until descriptor.elementsCount)
```

Comment:
It would be nice if we didn't have to build this `pairedPropertyNames` for every call to `decodeElementIndex` - is there any way to avoid this?

Reply:
I've changed it to use lateinit and initialise the first time `decodeElementIndex` is called in https://github.com/charleskorn/kaml/pull/400/commits/21d4295d43c93b32c6619611c83c6102ac28b054